src/Storages/MergeTree/TTLMergeSelector.cpp

```diff
@@ -73,12 +80,14 @@ IMergeSelector::PartsInPartition TTLMergeSelector::select(
         --best_begin;
     }
 
+    /// Find end of range with most old TTL.
     while (best_end != best_partition.end())
```

Comment:
Hmm, what if we are at the end of parts already? (meaning we only have a single part selected by TTLMergeSelector) Is it okay to return 1 part?

Reply:
Yes, it is okay, TTL merges can occur with 1 part (for 'thinning', if I used correct word for than).